app/scripts/controllers/network/createJsonRpcClient.js

```diff
@@ -36,26 +34,3 @@ export default function createJsonRpcClient({ rpcUrl, chainId }) {
 
   return { networkMiddleware, blockTracker };
 }
-
-function createChainIdMiddleware(chainId) {
```

Comment:
Could we move these changes to a separate PR?

Reply:
I agree, we shouldn't be making changes to any middleware in this PR, we should just add new tests.

Reply:
Leave a reaply

Reply:
I realized why some of these changes were made and in fact I think it is necessary to adjust this file a bit. It seems that the `eth_gasEstimate` RPC method has been modified, or rather decorated, in tests so that it runs 2 seconds slower. That said I don't think the middleware that is used to achieve this is written correctly, as it seems to pause the request forever. I amended this file in a more minimal way so that we could disable this behavior for the network client tests.

Reply:
Interesting. I'm having trouble understanding how your changes here resolved the issue. It appears to be functionally equivalent.

Reply:
This comment is tied to a previous commit, so is it possible you're not seeing the new changes? I can add a comment to the latest commit to explain what's happening.

Reply:
Ohh I see. IN_TEST is being changed dynamically. I hadn't considered that, I was used to the e2e context where it's converted to a string literal. But we can change it here.

Reply:
Fair enough! This seems like a pretty safe change to make, unlikely to have a functional impact.